gnd/src/main/java/com/google/android/gnd/ui/map/gms/GoogleMapsMapProvider.java

```diff
@@ -16,33 +16,41 @@
 
 package com.google.android.gnd.ui.map.gms;
 
+import android.content.Context;
 import androidx.fragment.app.Fragment;
 import com.google.android.gms.maps.GoogleMap;
-import com.google.android.gnd.persistence.local.LocalValueStore;
+import com.google.android.gnd.repository.MapsRepository;
 import com.google.android.gnd.rx.annotations.Hot;
 import com.google.android.gnd.ui.MarkerIconFactory;
 import com.google.android.gnd.ui.map.MapAdapter;
 import com.google.android.gnd.ui.map.MapProvider;
 import com.google.common.collect.ImmutableMap;
+import dagger.hilt.android.qualifiers.ApplicationContext;
 import io.reactivex.Single;
 import io.reactivex.subjects.SingleSubject;
 import java.util.HashMap;
 import java.util.Map;
+import javax.inject.Inject;
 
 /** Ground map adapter implementation for Google Maps API. */
 public class GoogleMapsMapProvider implements MapProvider {
 
+  private final Context context;
   private final MarkerIconFactory markerIconFactory;
-  private final LocalValueStore localValueStore;
+  private final MapsRepository mapsRepository;
```

Comment:
The `MapProvider` is intended as a wrapper around the map API and the application, to keep the implementation clean it shouldn't depend on any other application layers or state. Instead of injecting and passing `MapsRepository`, can `MapContainerFragment` set it instead in `onMapReady`?

Reply:
Done